code/userprofile/build.gradle.kts

```diff
@@ -0,0 +1,358 @@
+        }
+    }
+
+    publishing {
+        singleVariant("release") {
+            withSourcesJar()
+            withJavadocJar()
+        }
+    }
+
+    testOptions {
+        unitTests.isReturnDefaultValues = true
+    }
+
+    compileOptions {
+        sourceCompatibility = rootProject.extra["sourceCompatibility"] as JavaVersion
+        targetCompatibility = rootProject.extra["targetCompatibility"] as JavaVersion
+    }
+
+    kotlinOptions {
+        jvmTarget = rootProject.extra["kotlinJvmTarget"] as String
+        languageVersion = rootProject.extra["kotlinLanguageVersion"] as String
+        apiVersion = rootProject.extra["kotlinApiVersion"] as String
+    }
+}
+
+dependencies {
+    implementation("androidx.appcompat:appcompat:1.4.2")
+    implementation("com.adobe.marketing.mobile:core:2.0.1")
+
+    testImplementation("junit:junit:4.13.2")
+    testImplementation("org.mockito:mockito-core:4.5.1")
+    testImplementation("org.mockito:mockito-inline:4.5.1")
+    testImplementation("org.json:json:20180813")
+    testImplementation(
+        fileTree(
+            mapOf(
+                "dir" to "libs",
+                "include" to listOf("*.aar")
+            )
+        )
+    )
+    androidTestImplementation("androidx.test.ext:junit:1.1.3")
+    androidTestImplementation("androidx.test.espresso:espresso-core:3.4.0")
+}
+
+// Gradle Tasks
+tasks.withType<Test>().configureEach {
+    testLogging {
+        showStandardStreams = true
+    }
+    configure<JacocoTaskExtension> {
+        isIncludeNoLocationClasses = true
+        excludes = listOf("jdk.internal.*")
+    }
+}
+
+tasks.register<JacocoReport>("unitTestCoverageReport") {
+    dependsOn("testPhoneDebugUnitTest")
+    enableReports(enableHtmlReport = true, enableXmlReport = true, enableCsvReport = false)
+    addSourceSets()
+
+    executionData.setFrom(
+        fileTree(
+            mapOf(
+                "dir" to "$buildDir",
+                "includes" to listOf(BuildUtils.UNIT_TEST_EXECUTION_RESULTS_REGEX)
+            )
+        )
+    )
+}
+
+tasks.register<JacocoReport>("functionalTestsCoverageReport") {
+    dependsOn("createPhoneDebugCoverageReport")
+    enableReports(enableHtmlReport = true, enableXmlReport = true, enableCsvReport = false)
+    addSourceSets()
+
+    executionData.setFrom(
+        fileTree(
+            mapOf(
+                "dir" to "$buildDir",
+                "includes" to listOf(BuildUtils.FUNCTIONAL_TEST_EXECUTION_RESULTS_REGEX)
+            )
+        )
+    )
+}
+
+android.libraryVariants.configureEach {
+    tasks.withType(Javadoc::class.java).configureEach {
+        val mainSourceSet = android.sourceSets.getByName(BuildUtils.MAIN_SOURCE_SET)
+        val phoneSourceSet = android.sourceSets.getByName(BuildUtils.PHONE_SOURCE_SET)
+        val mainSourceDirs = mainSourceSet.java.srcDirs.first()
+        val phoneSourceDirs = phoneSourceSet.java.srcDirs.first()
```

Comment:
While this _works_ for our usecases because we have single source sets, I will change this to accumulate for multi source sets also.